src/Lykke.Service.PayInternal.AzureRepositories/Supervisor/SupervisorRepository.cs

```diff
@@ -14,55 +14,53 @@ namespace Lykke.Service.PayInternal.AzureRepositories.Supervisor
+        public async Task<ISupervisor> GetAsync(string employeeId)
         {
-            var entity = await _storage.GetDataAsync(
-                SupervisorEntity.IndexByEmployee.GeneratePartitionKey(merchantId),
-                SupervisorEntity.IndexByEmployee.GenerateRowKey(employeeId));
+            AzureIndex index = await _employeeIndexStorage.GetDataAsync(
+                SupervisorEntity.IndexByEmployee.GeneratePartitionKey(employeeId),
+                SupervisorEntity.IndexByEmployee.GenerateRowKey());
 
+            if (index == null) return null;
+            var entity = await _storage.GetDataAsync(index);
             return Mapper.Map<Core.Domain.Supervisor.Supervisor>(entity);
         }
 
         public async Task<ISupervisor> InsertAsync(ISupervisor supervisor)
         {
-            var entity = new SupervisorEntity(SupervisorEntity.IndexByEmployee.GeneratePartitionKey(supervisor.MerchantId), SupervisorEntity.IndexByEmployee.GenerateRowKey(supervisor.EmployeeId));
-            try
-            {
-                Mapper.Map(supervisor, entity);
-                await _storage.InsertAsync(entity);
-            }
-            catch (StorageException ex)
-            {
-                throw;
-            }
-
+            var entity = new SupervisorEntity(SupervisorEntity.IndexByEmployee.GeneratePartitionKey(supervisor.MerchantId),
+                SupervisorEntity.IndexByEmployee.GenerateRowKey(supervisor.EmployeeId));
+            Mapper.Map(supervisor, entity);
+            await _storage.InsertAsync(entity);
+            var index = SupervisorEntity.IndexByEmployee.Create(entity);
+            await _employeeIndexStorage.InsertAsync(index);
+            
             return entity;
         }
         public async Task<ISupervisor> UpdateAsync(ISupervisor supervisor)
         {
             var entity = new SupervisorEntity(SupervisorEntity.IndexByEmployee.GeneratePartitionKey(supervisor.MerchantId), SupervisorEntity.IndexByEmployee.GenerateRowKey(supervisor.EmployeeId));
-            try
-            {
-                entity.ETag = "*";
-                Mapper.Map(supervisor, entity);
-                await _storage.MergeAsync(entity.PartitionKey, entity.Id, e => entity);
-            }
-            catch (StorageException ex)
-            {
-                throw;
-            }
+            entity.ETag = "*";
+            Mapper.Map(supervisor, entity);
+            await _storage.MergeAsync(entity.PartitionKey, entity.Id, e => entity);
 
             return entity;
         }
-        public async Task DeleteAsync(string merchantId, string employeeId)
+        public async Task DeleteAsync(string employeeId)
         {
-            await _storage.DeleteAsync(SupervisorEntity.IndexByEmployee.GeneratePartitionKey(merchantId), SupervisorEntity.IndexByEmployee.GenerateRowKey(employeeId));
+            AzureIndex index = await _employeeIndexStorage.GetDataAsync(
```

Comment:
You are getting index from _employeeIndexStorage and deleting from _storage, really?
And what about deleting the entity itself?